core/src/main/java/org/apache/calcite/DataContext.java

```diff
@@ -122,7 +122,13 @@
      *
      * <p>Default value is "user.name" from
      * {@link System#getProperty(String)}. */
-    SYSTEM_USER("systemUser", String.class);
+    SYSTEM_USER("systemUser", String.class),
+
+    /** The sql mode.
+     *
+     * <p>Default value is "" from
+     * {@link System#getProperty(String)}. */
+    SQL_MODE("sqlMode", String.class);
```

Comment:
SQL mode appears to be a MySQL-specific thing. Could you explain why it's needed here?